### DISCUSSION TITLE
I can't update

### DISCUSSION THREAD

**Discussion by TomiTom1234** [erew123/alltalk_tts#121] [Q&A] [answered]
I tried to update and followed the steps mentioned here, which are:
-----------------
1- In a command prompt/terminal window you need to move into your alltalk_tts folder and run start_environment.bat or /start_environment.sh to load the Python environment.

2- At the command prompt/terminal, type git pull and wait for it to complete the download.
-----------------
I installed standalone version, but when I type the command `git pull` I get the following message in the command prompt: 
`fatal: not a git repository (or any of the parent directories): .git`

Any idea what I am doing wrong here?

**Comment by erew123** [comment]
Hi @TomiTom1234 

Within the AllTalk folder, there is a hidden folder `.git` 

![image](https://github.com/erew123/alltalk_tts/assets/35898566/889872d0-c48d-4220-a642-ea9219abea98)

Within this folder is all the tracking details it uses for doing anything with git, such as a `git pull`

If that folder is deleted, renamed or corrupted (and you are in the /alltalk_tts/ folder) then you would get the error. 

So below I did a git pull (ignore my message about committing changes, that's because I changed those files while working on code), but then I renamed the `.git` folder and got the error.

![image](https://github.com/erew123/alltalk_tts/assets/35898566/0eadc46e-0e9e-4c15-8a2a-dedb7b7b1d0f)

So I guess check that your `.git` folder is there, you have access to go inside it, as mentioned its a hidden folder.

I'm not sure if you are on Windows or Linux, if you are on Linux, I guess you could try a `sudo git pull` just to over-ride any possible permission issues.

Thats the first thing to be checking really.

**Reply by TomiTom1234** [reply]
Thank you for your reply.
I am on Windows.
I showed the hidden folders, but there is no hidden .git folder at all.
Do I have to just create one and do the update?

**Reply by erew123** [reply]
Hi @TomiTom1234 

No git folder... umm oh! I don't know how that could have happened.

The git folder has all the tracking of what current versions of files you have, where they are located, what files to copy over etc etc. You cant just create a folder and have it work. 

My best suggestion at this point it to rename your current `alltalk_tts` folder to `alltalk_tts.old` and then download and extract the current zip file in its place https://github.com/erew123/alltalk_tts/releases/download/1.9/alltalk_tts.zip

So you would have both an `alltalk_tts` folder and an `alltalk_tts.old`

From the `alltalk_tts.old` folder copy `models`, `outputs`, `voices` and `confignew.json` over, which should keep your settings in place. 

Not knowing how old your previous version is, you may want to re apply the requirements files.

Once you have it running and confirm all is good again, you can delete the `alltalk_tts.old` folder as/when you want. If there is some issue, you can always revert back.

In the newly extracted folder, you should be able to see the git folder now.

![image](https://github.com/erew123/alltalk_tts/assets/35898566/ba3289b5-0e06-455c-81f3-4d2caaed5f2e)

**Reply by erew123** [reply]
Oh, if you are on a standalone environment, you can either copy over the environment too `alltalk_environment` and your `start_alltalk.bat` as well as `start_environment.bat` or rebuild it with `atsetup.bat` 

![image](https://github.com/erew123/alltalk_tts/assets/35898566/b6e25ae6-9731-4bee-8cc4-a5dc73cd7130)

**Reply by TomiTom1234** [reply]
Thank you for your time and effort.
I did what you said, but this didn't create .git folder either.
I copied a .git folder from some other open-source generator (RVC) then pasted it in the Alltalk main directory, then tried the update commands, and it worked, but gave me the message that it is up-to-date. Furthermore, I don't know if that is OK.

**Reply by erew123** [reply]
Im not sure it will be ok. You will be up to date if you have extracted from that zip, but I dont think you will be pointing at the correct repository. Ive just updated the zip file again and pushed the git folder into it afresh. So you are welcome to pull the git folder out of that https://github.com/erew123/alltalk_tts/releases/download/1.9/alltalk_tts.zip

**Reply by TomiTom1234** [reply]
LOL I noticed  just now your reply, but weird enough that the .git folder isn't there when I extracted your zip file.
Does it run away from my machine when I just try to catch it? 😄

**Reply by erew123** [reply]
Hi @TomiTom1234 

Well I have a good news/bad news situation for you! Not that bad so don't worry!

During my last update, I spent around 2 days on documentation, part of which was to re-visit every question I've been asked on every platform. I've either done my best to code around issues OR test things out and write out help/instructions.

Bringing that back to your specific problem with updating a ZIp file copy, I've found out that even though the git folder may exist within the zip, it will refuse to update with a git pull. The only way to make it update with `git pull` updates, is to have git pulled the original copy.

The simplest method to move over to a version that has been git pulled will be to follow the quick setup instructions, then copy/migrate over the files as I showed above. All should be good then.

So good news, we have an answer, bad news I had lots of documentation to write and you will need to git pull next time you update.

Hope that solves it at least :)

**Comment by TomiTom1234** [comment]
Thanks man I appreciate your time and effort.

I am kinda lost here.. Just to clarify things:
The ZIP file you uploaded here for me to download, supposed to include ".git" folder or not? or does it get created after I replace my old files with yours then run `git pull` command?

Here is a screenshot of the your ZIP file contents, and as you can see no .git folder:
![Screenshot 2024-04-01 182434](https://github.com/erew123/alltalk_tts/assets/125768071/ce604d5a-3e73-4912-8972-55c3b7088aa3)

**Reply by erew123** [reply, marked as answer]
Hi @TomiTom1234 

So you would want to do the `git clone` when creating AllTalk, from the word go. Don't use the ZIP file unless you are building on a Headless system with NO internet access.

Although the zip file does contain the `github` folder, it appears this **doesn't** carry the tracking information to say that the files:

- are linked to `https://github.com/erew123/alltalk_tts/`
- are version number x, y, z etc

This information is only populated/created when you `git clone` the initial build of something onto your computer via the `git` utility. It builds a mini tracking database file that keeps track of all the file versions on your computer and compares them to the files stored up on Github.

So you will:

- `git clone https://github.com/erew123/alltalk_tts/` to download/create the `alltalk_tts` folder, then follow the instructions.

from then on you can:

- manually `git pull` when inside the `alltalk_tts` folder to pull down updates.
- Or use the `atsetup` utility to pull the updates, which is performing a git pull.

I have to say, git and github aren't the most friendly to your average person who just wants to dip into it, there can be too much to learn. Obviously I only recently had time to figure out that ZIP vs the `git clone` would work differently with `git pull` and how it populates the `.github` and `.git` folder.

So if you want to migrate to using a `git clone` method, you would:

1) Rename your current `alltalk_tts` folder to something like `alltalk_tts.old`.
2) Move to the folder where you want AllTalk to be inside of it e.g. `cd myfolder`
3) Perform the `git clone https://github.com/erew123/alltalk_tts/` which will pull AllTalk down, into a folder called `alltalk_tts` (as per the quick setup instructions).

---

![image](https://github.com/erew123/alltalk_tts/assets/35898566/57d79347-312d-4853-9122-5d575264c796)

---

4) You will of course need to run `atsetup` to install the relevant bits for Standalone OR Text-generation-webui. e.g.:

---

![image](https://github.com/erew123/alltalk_tts/assets/35898566/94e13c5c-37d7-40fc-8669-373ad1270e7e)

---

5) You can copy over the correct files/folders from your `alltalk_tts.old` folder to migrate the settings/files that you need.

`confignew.json`
`models`
`outputs`
`voices`

---

![image](https://github.com/erew123/alltalk_tts/assets/35898566/d217ab30-bdbe-40ee-b593-d1926844bf9b)

---

6) After you have done that and you can start up AllTalk.

Hope that clears it up.

Thanks

**Reply by erew123** [reply]
Oh, and assuming its working, you can delete the `alltalk_tts.old` folder at your leisure.

**Reply by TomiTom1234** [reply]
Thank you! I followed the steps and now everything is OK. There is finally a ".git" folder and everything is set and working.
